docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,228 @@ The remaining points can happen any time after 1) is complete although note that
+   functionality including optional configurations. See
+   [cylc-flow #3819](https://github.com/cylc/cylc-flow/issues/3819)
+   and [rose #2412](https://github.com/metomi/rose/pull/2412).
 
 6. Utility to remove installed locations (on all platforms) on demand
    (equivalent to `rose suite-clean`).
 
-7. On starting up a new run of the suite assign a new run name:
+7. Use a different run directory for each installation of a suite:
    ```
-   mi-aa001/
+   u-ab123/
      run1
      run2
    ```
-   * This is new functionality but Cylc 8 is a good time to introduce it.
+   * This is big change in working practise but Cylc 8 is a good time to
+     introduce it.
    * Use a symbolic link to identify the latest run which is the default name
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git.
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `run1` already exists it will install into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
+* `--run-name=my-run` implies install into `~/cylc-run/$(basename $PWD)/my-run`.
+  If the target directory already exists then fail.
+* `--no-run-name` implies install into `~/cylc-run/$(basename $PWD)`.
+* `--flow-name=my-flow` implies install into `~/cylc-run/my-flow/runN`.
+* `--directory=/path/to/flow` (`-C ...`) implies install the workflow found in
+  `/path/to/flow` (rather than `$PWD`).
+* All the options accepted by `cylc play` will be accepted by `cylc install`.
+  These options will provide the defaults for when `cylc play` is run with this
+  installed workflow (but they can still be overridden). Relevant options will
+  also apply to other cylc commands when operated on the installed workflow
+  (e.g. `cylc validate`). Note that supporting this will require the creation of
+  a new options file.
```

Comment:
I'm not convinced by this "install now with some options then run later" use case. Does anyone actually want to do this?

It is currently possible to "install with suite variables but not run" in Rose (`rose suite-run -l`) but this is seldom done. The only use case (beyond evaluating the result of running `rose suite-run`) that I'm aware of is for running "static analysis" commands like `validate` and `get-config`. With the retirement of `rose suite-run` these commands can be run directly against the source directory negating the need to install the workflow.

Note that we have talked about implementing an options file for Cylc in Python as an alternative to the `rose suite-conf` for specifying suite variables, etc (at Cylc workshops). So if we push ahead with this I would suggest approaching it from that angle. This file would provide good transparency to users (as it would be user-facing and could be defined in the src_dir) as well as supporting `src_dir == run_dir` uses.

Reply:
OK. I've removed this bit for the moment (we can always change our minds later).
I won't resolve it for the moment in case anyone else wants to comment.

Reply:
I'm fine with this. The only use case I'm aware of is to install then validate, with `rose suite-run`.